tools/GSDumpGUI/Library/GSDXWrapper.cs

```diff
@@ -205,115 +205,135 @@ public unsafe void Run(GSDump dump, int rendererOverride)
             fixed (byte* pointer = tempregisters)
             {
                 GSsetBaseMem(new IntPtr(pointer));
-                Int32 HWND = 0;
-                GSopen(new IntPtr(&HWND), "", rendererOverride);
+                IntPtr hWnd = IntPtr.Zero;
+
+                if (GSopen(new IntPtr(&hWnd), "", rendererOverride) != 0)
+                    return;
+
                 GSsetGameCRC(dump.CRC, 0);
 
+                NativeMethods.SetClassLong(hWnd,/*GCL_HICON*/ -14, Program.hMainIcon);
+
                 fixed (byte* freeze = dump.StateData)
                 {
-                    byte[] GSFreez = new byte[8];
-                    Array.Copy(BitConverter.GetBytes(dump.StateData.Length), 0, GSFreez, 0, 4);
-                    Array.Copy(BitConverter.GetBytes(new IntPtr(freeze).ToInt32()), 0, GSFreez, 4, 4);
+                    byte[] GSFreez;
+
+                    if (IntPtr.Size > 4)
```

Comment:
Here I would use Environment.is64bit to make it readable for people after you

Reply:
Done!